sirepo/template/template_common.py

```diff
@@ -208,13 +208,14 @@ def report_parameters_hash(data):
         models = sirepo.template.import_module(data).models_related_to_report(data)
         res = hashlib.md5()
         dm = data['models']
+        from six import string_types
         for m in models:
-            if isinstance(m, basestring):
+            if isinstance(m, str):
                 name, field = m.split('.') if '.' in m else (m, None)
                 value = dm[name][field] if field else dm[name]
             else:
                 value = m
-            res.update(json.dumps(value, sort_keys=True, allow_nan=False))
+            res.update(json.dumps(value, sort_keys=True, allow_nan=False).encode('utf-8'))
```

Comment:
The problem wasn't with `json.dumps()`, but with md5 hash, the error was similar to https://stackoverflow.com/q/7585307/4143531.

Reply:
We have to be careful here, because we should use the preferred encoding (decoding, really), because that's what json.dumps will write. You can't just pick utf-8. I think we want ensure_ascii=False, because that's going to be wrong. Or force the encoding on the dumps(encoding="utf-8"). The key is that you don't want an implicit coupling with utf-8, because it might not be that.

Reply:
`ensure_ascii=False` didn't solve the issue:

![screenshot-2018-4-17 nsls-ii chx beamline - srw - radiasoft](https://user-images.githubusercontent.com/13209176/38893776-50532700-4259-11e8-91a6-dcf74fd8c343.png)

Server's log:
```py
127.0.0.1 - - [17/Apr/2018 16:04:57] "POST /run-simulation HTTP/1.1" 200 -
../sirepo/runner.py:152:_start_job nI1MmB9d-Tcih54JF-brillianceReport: started: pid=21985 cmd=['sirepo', 'srw', 'run', '/Users/mrakitin/src/radiasoft/sirepo/run/user/nI1MmB9d/srw/Tcih54JF/brillianceReport']
../sirepo/server.py:1005:_simulation_error error: TypeError: Unicode-objects must be encoded before hashing
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/socketserver.py", line 639, in process_request_thread
    self.finish_request(request, client_address)
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/socketserver.py", line 361, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/socketserver.py", line 696, in __init__
    self.handle()
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/site-packages/werkzeug/serving.py", line 293, in handle
    rv = BaseHTTPRequestHandler.handle(self)
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/http/server.py", line 418, in handle
    self.handle_one_request()
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/site-packages/werkzeug/serving.py", line 328, in handle_one_request
    return self.run_wsgi()
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/site-packages/werkzeug/serving.py", line 270, in run_wsgi
    execute(self.server.app)
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/site-packages/werkzeug/serving.py", line 258, in execute
    application_iter = app(environ, start_response)
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/site-packages/flask/app.py", line 1997, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/site-packages/beaker/middleware.py", line 156, in __call__
    return self.wrap_app(environ, session_start_response)
  File "/Users/mrakitin/src/radiasoft/sirepo/sirepo/server.py", line 834, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/mrakitin/anaconda3/envs/sirepo_py3.6/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/mrakitin/src/radiasoft/sirepo/sirepo/uri_router.py", line 168, in _dispatch
    return route.func(**kwargs)
  File "/Users/mrakitin/src/radiasoft/sirepo/sirepo/server.py", line 531, in api_runSimulation
    res = _simulation_run_status(data)
  File "/Users/mrakitin/src/radiasoft/sirepo/sirepo/server.py", line 1040, in _simulation_run_status
    rep = simulation_db.report_info(data)
  File "/Users/mrakitin/src/radiasoft/sirepo/sirepo/simulation_db.py", line 652, in report_info
    rep.req_hash = template_common.report_parameters_hash(data)
  File "/Users/mrakitin/src/radiasoft/sirepo/sirepo/template/template_common.py", line 218, in report_parameters_hash
    res.update(json.dumps(value, sort_keys=True, allow_nan=False, ensure_ascii=False))
```